include/lsst/afw/geom/SkyWcs.h

```diff
@@ -0,0 +1,205 @@
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SKYWCS_H
+#define LSST_AFW_GEOM_SKYWCS_H
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "Eigen/Core"
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SpherePoint.h"
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/daf/base/PropertyList.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+/**
+Make a WCS CD matrix
+
+@param[in] scale  Pixel scale as an angle on sky/pixels
+@param[in] orientation  Position angle of focal plane +Y, measured from N through E.
+                        At 0 degrees, +Y is along N and +X is along E/W if flipX false/true
+                        At 90 degrees, +Y is along E and +X is along S/N if flipX false/true
+@param[in] flipX  Fip x axis (see orientation for details)?
+
+@return the CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+                        and i, j have range [1, 2]
+
+*/
+Eigen::Matrix2d makeCdMatrix(Angle const &scale, Angle const &orientation = 0 * degrees, bool flipX = false);
+
+/**
+A WCS that transform pixels to ICRS RA/Dec using the LSST standard for pixels
+
+The LSST standard is: 0,0 is the lower left pixel of the parent image
+(unlike FITS, which uses 1,1 as the lower left pixel of the subimage).
+
+The forward direction transforms from pixels (actual, if known, else nominal) to ICRS RA, Dec.
+
+@anchor skywcs_frames The contained ast::FrameSet must have the following frames.
+All are of class ast::Frame except the sky frame. The domain is listed first, if set:
+- ACTUAL_PIXEL0 (optional): actual pixels in LSST convention: 0,0 is lower left corner of parent image.
+    The 0 is to remind users that this frame is 0-based.
+- PIXEL0: nominal pixels in LSST convention: 0,0 is lower left corner of parent image
+- GRID: nominal pixels in FITS convention: 1,1 is lower left corner of subimage
+- IWC: intermediate world coordinates (the FITS WCS concept)
+- An ast::SkyFrame with System=ICRS and standard axis order RA, Dec.
+
+If ACTUAL_PIXEL0 is present then it will be the base frame, else PIXEL0 will be the base frame.
+The SkyFrame is always the current frame.
+
+Constructors must set the following properties of the contained ast::FrameSet:
+- Include the frames listed above, with specified domains.
+    Note that AST will provide the IWC frame when reading standard WCS FITS
+    data using an ast::FitsChan if you set "IWC=1" as a property of ast::FitsChan.
+    Other constructors may have to work harder.
+- For the output ast::SkyFrame set SkyRef=CRVAL (in radians) and SkyRefIs="Ignored".
+    This makes CRVAL easy to look up, and "Ignored" prevents CRVAL from being used as an offset.
+*/
+class SkyWcs : public Transform<Point2Endpoint, SpherePointEndpoint> {
```

Comment:
I don't think you need to worry about that. We will have some sort of design review before switching to this new class. I'm just hoping something of this code will remain afterwards.